modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/NetworkIdentifier.scala

```diff
@@ -21,6 +22,10 @@ case class NetworkIdentifier(
   network: NetworkEnvironment,
   subNetworkIdentifier: Option[SubNetworkIdentifier]
 )
+object NetworkIdentifier {
+  val Mainnet = NetworkIdentifier(BlockchainId.dag, NetworkEnvironment.Mainnet, none)
+  val Testnet = NetworkIdentifier(BlockchainId.dag, NetworkEnvironment.Testnet, none)
```

Comment:
we already have `NetworkEnvironment.fromAppEnvironment` so I think it's redundant to hardcode specific environments here. you may add `NetworkIdentifier.fromAppEnvironment` that would return you the correct identifier

Reply:
`NetworkIdentifier.fromAppEnvironment` returns an `Option` which means I need to add runtime logic to handle a case that will never happen because we know what we want to respond with at compile time. So a constant makes more sense to me.

Reply:
yes, it returns `Option` because you can launch an environment that does not map into network identifier.
having it constant, you would just return invalid data

Reply:
so it's not a case that will never happen. you compile a single version and you specify the environment when running the node, not at compile time.

Reply:
If I write `NetworkIdentifier.fromAppEnvironment`

1. What is the return type? If it returns an `Option` then my what about my comment above?
2. We have 3 `AppEnvironment`s defined but only 2 are mapped to a `NetworkEnvironment` -- `Dev` is excluded. Am I to do the same thing for `NetworkIdentifier.fromAppEnvironment` ?
3. If we don't map `Dev`, what should the endpoint return for that environment? Empty list?

Reply:
re.1: Which network identifier you want to return on development environment?
re.2: up to you (you can map every app environment if you want, depending where rosetta should work)
re.3: I have no idea but returning either mainnet or testnet, or both, seems to be invalid, so imo there is no point in hardcoding network identifiers

Reply:
Refactored. The code now (once I push it) defines the network id in terms of the network env and will return an empty list for the Dev environment since that's not defined.